docs/externalGrading.md

```diff
@@ -258,44 +258,51 @@ In production, PrairieLearn runs external grading jobs on a distributed system t
 - We need to get job files from inside the Docker container running PrairieLearn to the host machine so that Docker can mount them to `/grade` on the grading machine. We achieve this by mounting a directory on the host machine to `/jobs` on the grading machine, and setting an environment variable `HOST_JOBS_DIR` containing the absolute path of that directory on the host machine.
 
-Running PrairieLearn locally with externally graded question support looks something like this:
-
-- Create an empty directory to use to share job data between containers.
-  - This can live anywhere, but needs to be created first and referenced in
-    the docker launch command.
-  - This command is copy-pastable for Windows PowerShell, MacOS, and Linux (including WSL2 instances).
+To run PrairieLearn locally with externally graded question support, create an empty directory to use to share job data between containers. This directory can live anywhere, but needs to be created first and referenced in the docker launch command. This directory only needs to be created once. If you are running Windows with WSL 2, the directory should be created inside the WSL 2 instance. You can create this directory using a command like:
 
 ```bash
 mkdir "$HOME/pl_ag_jobs"
 ```
 
-- Modify your PL docker run call to include the jobs directory.
+Now, run PrairieLearn as usual, but with additional options. For example, if your course directory is in `$HOME/pl-tam212` and the jobs directory created above is in `$HOME/pl_ag_jobs`, and you are using Linux or Mac OS X, the new command is as follows:
 
-##### Shell Command
+```sh
+docker run -it --rm -p 3000:3000 \
+    -v "$HOME/pl-tam212:/course" `# Replace the path with your course directory` \
+    -v "$HOME/pl_ag_jobs:/jobs" `# Map jobs directory into /jobs` \
+    -e HOST_JOBS_DIR="$HOME/pl_ag_jobs" \
+    -v /var/run/docker.sock:/var/run/docker.sock `# Mount docker into itself so container can spawn others` \
+    prairielearn/prairielearn
+```
 
-On MacOS and Linux, `cd` to your course directory and copy-paste the following command:
+If you are on Windows with WSL 2, you can use the following command on the WSL 2 shell:
 
 ```sh
 docker run -it --rm -p 3000:3000 \
-    -v "$PWD":/course `# Map your current directory in as course content` \
+    -v "$HOME/pl-tam212:/course" `# Replace the path with your course directory` \
     -v "$HOME/pl_ag_jobs:/jobs" `# Map jobs directory into /jobs` \
     -e HOST_JOBS_DIR="$HOME/pl_ag_jobs" \
     -v /var/run/docker.sock:/var/run/docker.sock `# Mount docker into itself so container can spawn others` \
+    --add-host=host.docker.internal:172.17.0.1 \
     prairielearn/prairielearn
 ```
 
-On Windows PowerShell, `cd` to your course directory and copy the following command **except** with your own username in `HOST_JOBS_DIR`:
+#### Running on Windows without WSL 2 support
```

Comment:
Is it even worth documenting this? It seems like running without WSL 2 is just a horrific experience.

Reply:
> Is it even worth documenting this? It seems like running without WSL 2 is just a horrific experience.

I thought about that, and tried to err on the side of keeping it just in case, but indeed for external graders and workspaces the experience can be quite deceiving. I'll remove it and indicate that we don't support the use of Windows without WSL 2 when using external graders and workspaces.

Reply:
I think that's a good idea, but I still wonder if we should even document it for running "just" PrairieLearn. Per the linked discussion, it took almost four minutes to boot everything on a very beefy machine. I can't imagine anyone wanting that when WSL exists.

Reply:
Done in 4c320bba6.